tests/test_generators/test_graphql.py

```diff
@@ -0,0 +1,85 @@
+import pytest
+
+from linkml.generators.graphqlgen import GraphqlGenerator
+
+PERSON = """
+type Person implements HasAliases
+  {
+    id: String!
+    name: String
+    hasEmploymentHistory: [EmploymentEvent]
+    hasFamilialRelationships: [FamilialRelationship]
+    hasMedicalHistory: [MedicalEvent]
+    ageInYears: Integer
+    addresses: [Address]
+    hasBirthEvent: BirthEvent
+    speciesName: String
+    stomachCount: Integer
+    isLiving: LifeStatusEnum
+    aliases: [String]
+  }
+"""
+
+MEDICALEVENT = """
+type MedicalEvent
+  {
+    startedAtTime: Date
+    endedAtTime: Date
+    isCurrent: Boolean
+    metadata: AnyObject
+    inLocation: Place
+    diagnosis: DiagnosisConcept
+    procedure: ProcedureConcept
+  }
+"""
+
+FAMILIALRELATIONSHIP = """
+type FamilialRelationship
+  {
+    startedAtTime: Date
+    endedAtTime: Date
+    cordialness: String
+    type: FamilialRelationshipType!
+    relatedTo: Person!
+  }
+"""
+
+DATASET = """
+type Dataset
+  {
+    metadata: AnyObject
+    persons: [Person]
+    companies: [Company]
+    activities: [Activity]
+    codeSystems: [CodeSystem]
+  }
+"""
+
+FAMILIALRELATIONSHIPTYPE = """
+enum FamilialRelationshipType
+  {
+    SIBLING_OF
+    PARENT_OF
+    CHILD_OF
+  }
+"""
+
+
+@pytest.mark.parametrize(
+    "input_class,expected",
+    [
+        # check that expected GraphQL schema blocks are present
+        ("Person", PERSON),
+        ("Dataset", DATASET),
+        ("MedicalEvent", MEDICALEVENT),
+        ("FamilialRelationship", FAMILIALRELATIONSHIP),
+        ("FamilialRelationshipType", FAMILIALRELATIONSHIPTYPE),
+    ],
+)
+def test_serialize_selected(input_class, expected, kitchen_sink_path):
```

Comment:
I am in general not a huge fan of string matching tests, as they are not specific and require a decent amount of maintenance to keep up to date. If we did want to leave these as string comparisons, we should use the snapshot system so that it's possible to update them when the generator changes. 

Otherwise, I think these should be split up into separate tests that test for specific things, even if those specific things are as general as "the graphql generator generates enums from schemas" and so on. From this test, I am not sure what metamodel features the graphql generator is supposed to support, and it's not clear to me what it's testing (well, it is because there are few cases, but as this grows that will become less true).

I also think we would really benefit from more unit-like tests and less end-to-end tests. Something like this that directly invokes the method rather than only testing the final output:

```python
def test_visit_enum():
    an_enum = EnumDefinition(...)
    generator = GraphqlGenerator()
    generated = generator.visit_enum(an_enum)
    
    assert all([v in generated for v in an_enum.permissible_values])
    # and hopefully an actual test of the syntax...
```

which is incomplete but you get what i mean.

Ideally we also directly test the correctness of the models by rigging ourselves up some test fixtures that actually create a graphql db and seed it with some data so we can test valid/invalid values in enums, that they actually can be used and so on, because substring matching leaves us open to a zillion invisible bugs from things as simple as string formatting, but idrc, i don't use graphql, am not core maintainer, and would be a decent amount of work for this PR so obviously i wouldn't insist on it here.

Reply:
@sneakers-the-rat this is exactly what the compliance tests are for.. but I think we are overloading this PR a bit...

Reply:
Testing based on snapshots also added on later changes.